appengine/standard_python37/pubsub/main.py

```diff
@@ -57,13 +61,31 @@ def index():
 # [START push]
 @app.route('/_ah/push-handlers/receive_messages', methods=['POST'])
 def receive_messages_handler():
-    # Verify that the request originates from the application.
+    # Verify the request originates from the application.
     if (request.args.get('token', '') !=
             current_app.config['PUBSUB_VERIFICATION_TOKEN']):
         return 'Invalid request', 400
 
-    # TODO: I need add some code to verify the bearer token too.
-
+    # Verify the push request originates from Cloud Pub/Sub.
+    try:
+        # Get the OpenIDConnect JWT in the "Authorization" header
+        # attached to the push request by Cloud Pub/Sub.
+        bearer_token = request.headers.get('Authorization')
+        token = bearer_token.split(' ')[1]
+        TOKENS.append(token)
+
+        header = jwt.decode_header(token)
+        HEADERS.append(header)
+
+        # Verify and decode the token. Underneath it checks the signature
+        # with the signed section using Google's public certs at
+        # https://www.googleapis.com/oauth2/v1/certs
+        req = requests.Request()
+        claim = id_token.verify_oauth2_token(token, req)
+        CLAIMS.append(claim)
+    except Exception as e:
+        CLAIMS.append(e)
+        return 'Unable to verify: ' + e, 400
```

Comment:
This line will result in a `TypeError`. In order to avoid it, the exception object must first be converted to a string before concatenation.